lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +705,871 @@ impl EnsoLexer {
+        ));
+        rule!(root_state(format_block_quote) => self.text_on_inline_block(
+            token::TextStyle::FormatInlineBlock,
+            self.text_format_inline_block,
+            reader
+        ));
+        rule!(root_state(invalid_raw_quote) => self.on_invalid_quote(reader));
+        rule!(root_state(raw_quote)         => self.on_begin_raw_line(reader));
+        rule!(root_state(raw_block_lf) => self.on_begin_text_block(
+            token::TextStyle::RawBlock,
+            self.text_raw_block,
+            token::LineEnding::LF,
+            reader
+        ));
+        rule!(root_state(raw_block_crlf) => self.on_begin_text_block(
+            token::TextStyle::RawBlock,
+            self.text_raw_block,
+            token::LineEnding::CRLF,
+            reader
+        ));
+        rule!(root_state(raw_block_quote) => self.text_on_inline_block(
+            token::TextStyle::RawInlineBlock,
+            self.text_raw_inline_block,
+            reader
+        ));
+
+
+        // === Rules for Handling Lines in Text Blocks ===
+
+        let in_block_line_id = lexer.in_block_line;
+        let in_block_line    = lexer.group_mut(in_block_line_id);
+        rule!(in_block_line(format_block_lf) => self.on_begin_text_block(
+            token::TextStyle::FormatBlock,
+            self.text_format_block,
+            token::LineEnding::LF,
+            reader
+        ));
+        rule!(in_block_line(format_block_crlf) => self.on_begin_text_block(
+            token::TextStyle::FormatBlock,
+            self.text_format_block,
+            token::LineEnding::CRLF,
+            reader
+        ));
+        rule!(in_block_line(raw_block_lf) => self.on_begin_text_block(
+            token::TextStyle::RawBlock,
+            self.text_raw_block,
+            token::LineEnding::LF,
+            reader
+        ));
+        rule!(in_block_line(raw_block_crlf) => self.on_begin_text_block(
+            token::TextStyle::RawBlock,
+            self.text_raw_block,
+            token::LineEnding::CRLF,
+            reader
+        ));
+
+
+        // === Escape Sequences ===
+
+        let text_escape_id = lexer.text_escape;
+        let text_escape    = lexer.group_mut(text_escape_id);
+        for char_escape in escape::EscapeCharacter::codes() {
+            let pattern = Pattern::all_of(&char_escape.pattern);
+            let call    = format!("self.on_escape_literal({:?},reader)",&char_escape.repr.as_str());
+            text_escape.create_rule(&pattern,call.as_str());
+        }
+        rule!(text_escape(escape_byte)    => self.on_escape_byte(reader));
+        rule!(text_escape(escape_u21)     => self.on_escape_u21(reader));
+        rule!(text_escape(escape_u16)     => self.on_escape_u16(reader));
+        rule!(text_escape(escape_u32)     => self.on_escape_u32(reader));
+        rule!(text_escape(escape_slash)   => self.on_escape_slash(reader));
+        rule!(text_escape(escape_invalid) => self.on_escape_invalid(reader));
```

Comment:
Please do 2 things:

1. rename `rule` to `rules`
2. Allow it to define multiple rules in a block :)